pkg/apis/cluster/v1alpha1/machinedeployment_types_test.go

```diff
@@ -0,0 +1,199 @@
+			// setup the test scenario
+			ctx := genericapirequest.NewDefaultContext()
+			target := v1alpha1.MachineDeploymentValidationStrategy{}
+
+			// act
+			errors := target.Validate(ctx, test.machineDeploymentToTest)
+
+			// validate
+			if len(errors) > 0 && !test.expectError {
+				t.Fatalf("an unexpected error was returned = %v", errors)
+			}
+			if test.expectError && len(errors) == 0 {
+				t.Fatal("expected an error but non was returned")
+			}
+
+		})
+	}
+
+}
+
+func crudAccessToMachineDeploymentClient(t *testing.T, cs *clientset.Clientset) {
+	instance := v1alpha1.MachineDeployment{
+		Spec: v1alpha1.MachineDeploymentSpec{
+
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"foo": "bar"},
+			},
+			Template: v1alpha1.MachineTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"foo": "bar"},
+				},
+			},
+		},
+	}
+	instance.Name = "instance-1"
+
+	expected := instance.DeepCopy()
+	// Defaulted fields.
+	var replicas int32 = 1
+	expected.Spec.Replicas = &replicas
+	var deadline int32 = 600
+	expected.Spec.ProgressDeadlineSeconds = &deadline
+	var limit int32 = 1
+	expected.Spec.RevisionHistoryLimit = &limit
+	expected.Spec.Strategy.Type = common.RollingUpdateMachineDeploymentStrategyType
+	unavailable := intstr.FromInt(0)
+	surge := intstr.FromInt(1)
+	rollingUpdate := v1alpha1.MachineRollingUpdateDeployment{
+		MaxUnavailable: &unavailable,
+		MaxSurge: &surge,
+	}
+	expected.Spec.Strategy.RollingUpdate = &rollingUpdate
+
+	// When sending a storage request for a valid config,
+	// it should provide CRUD access to the object.
+	client := cs.ClusterV1alpha1().MachineDeployments(metav1.NamespaceDefault)
+
+	// Test that the create request returns success.
+	actual, err := client.Create(&instance)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Delete(instance.Name, &metav1.DeleteOptions{})
+	if !reflect.DeepEqual(actual.Spec, expected.Spec) {
+		t.Fatalf(
+			"Default fields were not set correctly.\nActual:\t%+v\nExpected:\t%+v",
+			actual.Spec, expected.Spec)
+	}
+
+	// Test getting the created item for list requests.
+	result, err := client.List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if itemLength := len(result.Items); itemLength != 1 {
+		t.Fatalf("Number of items in Items list should be 1, but is %d.", itemLength)
+	}
+	if resultSpec := result.Items[0].Spec; !reflect.DeepEqual(resultSpec, expected.Spec) {
+		t.Fatalf(
+			"Item returned from list is not equal to the expected item.\nActual:\t%+v\nExpected:\t%+v",
+			resultSpec, expected.Spec)
+	}
+
+	// Test getting the created item for get requests.
+	actual, err = client.Get(instance.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(actual.Spec, expected.Spec) {
+		t.Fatalf(
+			"Item returned from get is not equal to the expected item.\nActual:\t%+v\nExpected:\t%+v",
+			actual.Spec, expected.Spec)
+	}
+
+	// Test deleting the item for delete requests.
```

Comment:
Does this comment belong bellow with the actual delete request?

Reply:
moved comment closer to delete request